src/data/list/basic.lean

```diff
@@ -1362,6 +1395,16 @@ theorem map_filter_eq_foldr (f : α → β) (p : α → Prop) [decidable_pred p]
   map f (filter p as) = foldr (λ a bs, if p a then f a :: bs else bs) [] as :=
 by { induction as, { refl }, { simp! [*, apply_ite (map f)] } }
 
+lemma filter_eq {α} (p q : α → Prop) [decidable_pred p] [decidable_pred q] (l : list α)
```

Comment:
This is [`list.filter_congr`](https://github.com/leanprover-community/mathlib/blob/265c587/src/data/list/basic.lean#L2594), right?